js/ui/calendar/ui.calendar.base_view.js

```diff
@@ -253,36 +278,38 @@ const BaseView = Widget.inherit({
         return this._$table.find(`.${CALENDAR_CONTOURED_DATE_CLASS}`);
     },
 
-    _changeValue: function(cellDate) {
-        if(cellDate) {
-            const value = this.option('value');
-            const newValue = value ? new Date(value) : new Date();
-
-            newValue.setDate(cellDate.getDate());
-            newValue.setMonth(cellDate.getMonth());
-            newValue.setFullYear(cellDate.getFullYear());
-            newValue.setDate(cellDate.getDate());
+    _renderValue: function() {
+        if(!this.option('allowValueSelection')) {
+            return;
+        }
 
-            this.option('value', newValue);
-        } else {
-            this.option('value', null);
+        let value = this.option('value');
+        if(!Array.isArray(value)) {
+            value = [value];
         }
+
+        this._$selectedCells?.forEach((cell) => { cell.removeClass(CALENDAR_SELECTED_DATE_CLASS); });
+        this._$selectedCells = value.map((value) => this._getCellByDate(value));
+        this._$selectedCells.forEach((cell) => { cell.addClass(CALENDAR_SELECTED_DATE_CLASS); });
```

Comment:
We can use a function to reduce a copypaste
 ```javascript
        _toggleSelectedClass($cell, isSelected) {
            $cell?.toggleClass(CALENDAR_SELECTED_DATE_CLASS, isSelected);
        }
```
We can use it in ~8 rows, check all places plz

Reply:
there 4 different classes, so I added `_toggleClass($cell, className, isSelected)